File: Semester 1/ALGOLAB1_Python/taskF.py
def main():
    n = int(input())
    numbers = list(map(int, input().split()))

    main_sum = sum(numbers)

    sums_left = 0
    sums_right = main_sum

    for i in range(n):
        if i == 0:
            sums_left = 0
            sums_right = main_sum - numbers[i]
        elif i == n - 1:
            sums_left = main_sum - numbers[i]
            sums_right = 0
        else:
            sums_left += numbers[i - 1]
            sums_right -= numbers[i]

        if sums_left == sums_right:
            print(i)
            return

    print(-1)

if __name__ == "__main__":
    main()

File: Semester_1/ALGOLAB1/taskF.cpp
#include <iostream>

using namespace std;

int main() {

  int n, mainSum = 0;

  bool flag = true;
  cin >> n;
  int numbers[n];
  for (int i = 0; i < n; i++) {
    cin >> numbers[i];
  }

  for (int i = 0; i < n; i++) {
    mainSum += numbers[i];
  }

  int sumsLeft;
  int sumsRight;
  sumsLeft = 0;
  sumsRight = mainSum;

  for (int i = 0; i < n; i++) {
    if (i == 0) {
      sumsLeft = 0;
      sumsRight = mainSum - numbers[i];
    }

    if (i == n - 1) {
      sumsLeft = mainSum - numbers[i];
      sumsRight = 0;
    }

    if (i != 0 and i != n - 1) {
      sumsLeft += numbers[i - 1];
      sumsRight -= numbers[i];
    }

    if (sumsLeft == sumsRight) {
      cout << i;
      flag = false;
      break;
    }
  }
  if (flag) {
    cout << -1;
  }
  return 0;
}